common/src/main/java/dev/ftb/mods/ftbchunks/client/FTBChunksClient.java

```diff
@@ -94,1187 +133,1202 @@
+            return EventResult.interruptTrue();
+        } else if (doesKeybindMatch(toggleMinimapKey, keyCode, scanCode, modifiers)) {
+            FTBChunksClientConfig.MINIMAP_ENABLED.set(!FTBChunksClientConfig.MINIMAP_ENABLED.get());
+            FTBChunksClientConfig.saveConfig();
+            return EventResult.interruptTrue();
+        } else if (doesKeybindMatch(openClaimManagerKey, keyCode, scanCode, modifiers)) {
+            try {
+                ChunkScreen.openChunkScreen();
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+            return EventResult.interruptTrue();
+        } else if (doesKeybindMatch(zoomInKey, keyCode, scanCode, modifiers)) {
+            return changeZoom(true);
+        } else if (doesKeybindMatch(zoomOutKey, keyCode, scanCode, modifiers)) {
+            return changeZoom(false);
+        } else if (doesKeybindMatch(addWaypointKey, keyCode, scanCode, modifiers)) {
+            return addQuickWaypoint();
+        } else if (doesKeybindMatch(waypointManagerKey, keyCode, scanCode, modifiers)) {
+            new WaypointEditorScreen().openGui();
+            return EventResult.interruptTrue();
+        }
+
+        return EventResult.pass();
+    }
+
+    public EventResult keyPressed(Minecraft client, Screen screen, int keyCode, int scanCode, int modifiers) {
+        if (doesKeybindMatch(openMapKey, keyCode, scanCode, modifiers) && Platform.isFabric()) {
+            // platform specific behaviour :(  why? ¯\_(ツ)_/¯
+            LargeMapScreen gui = ClientUtils.getCurrentGuiAs(LargeMapScreen.class);
+            if (gui != null && !gui.anyModalPanelOpen()) {
+                gui.closeGui(false);
+                return EventResult.interruptTrue();
+            }
+        }
+
+        return EventResult.pass();
+    }
+
+    private EventResult addQuickWaypoint() {
+        StringConfig name = new StringConfig();
+        Player player = Minecraft.getInstance().player;
+        if (player == null) return EventResult.pass();
+
+        return MapManager.getInstance().map(manager -> {
+            BaseScreen screen = new WaypointAddScreen(name, player);
+            screen.openGuiLater();
+            // later needed to prevent keypress being passed into gui
+            return EventResult.interruptTrue();
+        }).orElse(EventResult.pass());
+    }
+
+    private EventResult changeZoom(boolean zoomIn) {
+        prevZoom = FTBChunksClientConfig.MINIMAP_ZOOM.get();
+        double zoom = prevZoom;
+        double zoomFactor = zoomIn ? 1D : -1D;
+
+        if (zoom + zoomFactor > 4D) {
+            zoom = 4D;
+        } else if (zoom + zoomFactor < 1D) {
+            zoom = 1D;
+        } else {
+            zoom += zoomFactor;
+        }
+
+        lastZoomTime = System.currentTimeMillis();
+        FTBChunksClientConfig.MINIMAP_ZOOM.set(zoom);
+        return EventResult.interruptTrue();
+    }
+
+    private static long clamp(long val, long min, long max) {
```

Comment:
Likely not a you issue as this code has just been reformatted but this should just use the `Mth` class